lib/grammers-mtsender/src/lib.rs

```diff
@@ -128,13 +136,74 @@ impl Enqueuer {
 }
 
 impl<T: Transport, M: Mtp> Sender<T, M> {
-    async fn connect<A: ToSocketAddrs>(
+    async fn connect(
         transport: T,
         mtp: M,
-        addr: A,
+        addr: SocketAddr,
     ) -> Result<(Self, Enqueuer), io::Error> {
         info!("connecting...");
-        let stream = TcpStream::connect(addr).await?;
+
+        let stream: Box<dyn Stream> = match std::env::var("GRAMMERS_CLIENT_PROXY") {
```

Comment:
This will need to change before I can merge it. I propose we have a separate method, called something like `connect_with_proxy`, which is just like `connect`, but also accepts the proxy URL.